### PR TITLE
[camera_web] Add `availableCameras` implementation

### PR DESCRIPTION
**Pull request by bselwe**
Adds `availableCameras` implementation of the camera platform interface.

- Added `MediaDeviceKind` - represents a kind of a media device (video input, audio input, audio output). Used when filtering video input devices in `availableCameras`.
- Added `CameraSettings` - a utility to fetch and map camera settings. Used to extract the camera lens direction from the video track. Currently supports the following methods:
  - getFacingModeForVideoTrack
  - mapFacingModeToLensDirection
- Added `CameraMetadata` used along the `CameraDescription` to store additional web-specific camera details. They are going to be used when creating a camera later.
- Added `availableCameras` implementation.
- ~~Temporarily added `camera_web` to camera example pubspec.~~ (Reverted by @ditman, this dependency can be added to any app needing this package directly, not necessarily to the `camera` plugin)

Part of https://github.com/flutter/flutter/issues/45297.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I [updated pubspec.yaml](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates) with an appropriate new version according to the [pub versioning philosophy].
- [ ] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code

### CODE REVIEW

**Comment by bselwe** [comment]
@ditman @felangel @orestesgaolin

File: packages/camera/camera/example/pubspec.yaml [hunk 1]
```diff
@@ -14,6 +14,13 @@ dependencies:
     # The example app is bundled with the plugin so we use a path dependency on
     # the parent directory to use the current plugin's version.
     path: ../
+  
+  # Temporarily include a web implementation of the camera plugin
+  # as it is not yet integrated into the official package.
+  # TODO(bselwe): Remove when camera_web is integrated into the camera plugin.
+  camera_web:
+    path: ../../camera_web/
```
Comment (ditman) [inline packages/camera/camera/example/pubspec.yaml:22]: We cannot merge with this dependency here (people can add it themselves when using this package from Git)

File: packages/camera/camera_web/example/integration_test/types/camera_metadata_test.dart [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:camera_web/src/types/types.dart';
+import 'package:flutter_test/flutter_test.dart';
```
Comment (ditman) [inline packages/camera/camera_web/example/integration_test/types/camera_metadata_test.dart:6]: This test can be moved to camera_web/test and run as a VM test, no need to run this as an integration test.

**Review by ditman — APPROVED** [review]
This is looking good, I'm going to try and implement my own comments for this PR so we can keep going :)

**Comment by ditman** [comment]
@google I consent.

File: packages/camera/camera_web/example/integration_test/types/camera_metadata_test.dart (hunk 2 above)
Comment (bselwe) [reply packages/camera/camera_web/example/integration_test/types/camera_metadata_test.dart]: Good point!

**Pull request merged**